extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/QuarkusRestPathTemplateInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkus.resteasy.runtime;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import io.quarkus.arc.ArcInvocationContext;
+import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
+import io.vertx.core.http.impl.HttpServerRequestInternal;
+
+@SuppressWarnings("unused")
+@QuarkusRestPathTemplate
+@Interceptor
+@Priority(Interceptor.Priority.LIBRARY_BEFORE + 10)
+public class QuarkusRestPathTemplateInterceptor {
+    @Inject
+    CurrentVertxRequest request;
+
+    @AroundInvoke
+    Object restMethodInvoke(InvocationContext context) throws Exception {
+        QuarkusRestPathTemplate annotation = getAnnotation(context);
+        try {
+            return context.proceed();
+        } finally {
+            if (annotation != null) {
+                ((HttpServerRequestInternal) request.getCurrent().request()).context().putLocal("UrlPathTemplate",
```

Comment:
This needs to happen before the call, as async requests can complete before this method has actually returned. E.g. if you have the following code you will see the response being completed by the new thread before this call stack has returned:

```
    @GET
    @Path("gauge/{number}")
    public void checkListSize(@PathParam("number") long number, @Suspended AsyncResponse ar) throws Exception{
        if (number == 2 || number % 2 == 0) {
            // add even numbers to the list
            list.add(number);
        } else {
            // remove items from the list for odd numbers
            try {
                number = list.removeFirst();
            } catch (NoSuchElementException nse) {
                number = 0;
            }
        }
        long res = number;
        new Thread(new Runnable() {
            @Override
            public void run() {
                ar.resume(res);
            }
        }).start();
        Thread.sleep(1000);
    }
```